### ISSUE TITLE
Autocomplete on SObjects

### ISSUE THREAD

**Issue by ChuckJonas** [forcedotcom/salesforcedx-vscode#575]
### Summary

When Autocomplete is run on an SObject, it returns Global objects (EG `System`, `ApexPages`), which are no valid given the current instance context.   It also only returns a very limited set of fields

### Steps To Reproduce:

1.  Create new class call testing
2.  Add the following code:

```java
public class Testing {
    public Testing() {
        Account acc = new Account();
        acc.
    }
}
```
3. Trigger a completion request on `acc.` (`cmd+.`)

### Expected result

I would see the SObject Instance methods and all of it's fields (including custom fields)

### Actual result

I see the SObject instance methods, the global System namespaces & and very limited set of fields.

<img width="403" alt="testing_cls_ _realself" src="https://user-images.githubusercontent.com/5217568/44371077-543cf500-a49a-11e8-873c-ce27ba60ec7e.png">

### Additional information

_Feel free to attach a screenshot_.

**VS Code Version**:
1.26.0

**SFDX CLI Version**:
salesforcedx@pre-release

**OS and version**:
osx 10.12.6

**Comment by ChuckJonas** [comment]
Also, you get similar behavoir if completions are requested @ `new Account(`.  I would expect this to only give Account fields.

**Comment by vazexqi** [comment]
@ChuckJonas  - Did you do a "SFDX: Refresh SObject Defintions" after you have connected to an org? See https://github.com/forcedotcom/salesforcedx-vscode/tree/develop/packages/salesforcedx-vscode-apex#enable-code-smartness-for-sobjects for more information. 

We need to do an `sobjectdescribe` on the org to get the information – yes, this deviates a bit from our SFDX model but the org currently still has the most information about the standard objects, custom fields, and custom objects.

**Comment by ntotten** [comment]
Similar issue reported on #399  for string literals

When typing inside of a String literal, if I complete a sentence with a ".", it pulls up the intellisense dropdown, which I have to press escape on in order to continue typing.

![screen shot 2018-04-25 at 4 01 24 pm](https://user-images.githubusercontent.com/525742/39269864-128fc8ac-48a2-11e8-9c88-03a00b3fd599.png)

**Comment by ChuckJonas** [comment]
@ntotten is all of this happening inside the close source jar?  

Also if you guys haven't already taken a look at what @neowit has done with [tooling-force](https://github.com/neowit/tooling-force.com) it might be worth while peaking in.  IMO the autocomplete features has implemented here are the most robust (been a while since I've used illuminated cloud or welkin).   Supports completion in SOQL, SObject creation, attributes, inner classes.  No need for proxy SObject classes.  Etc

My [Salesforce Language Support](https://marketplace.visualstudio.com/items?itemName=chuckjonas.apex-autocomplete) extension is pretty much just a wrapper of this library.  Unfortunately there are some limitations around memory and currently no support for cancelation tokens so it doesn't work great on larger orgs.

**Comment by vazexqi** [comment]
Thanks for the link, @ChuckJonas. I am aware of it but due to its [LGPL v3 licensing](https://github.com/neowit/tooling-force.com#legal-stuff), we are very careful not to use/link it.

**Comment by ntotten** [comment]
@ChuckJonas Correct, these fixes are all happing in the closed source jar. We have discussed open sourcing it in the future, but it's not something we're able to do just yet.
